### PR TITLE
Add corrected citation.cff

### PR DESCRIPTION
**Pull request by diehlpk**
This should add the JOSS paper to the citation function.

### CODE REVIEW

**Comment by diehlpk** [comment]
@hkaiser I checked the documentation, and we can add the Zenodo and some references, but not the references only. I will update the pull request tomorrow.

Here, Is one example for the hpxcl repo. 

https://github.com/STEllAR-GROUP/hpxcl/blob/main/CITATION.cff

**Comment by diehlpk** [comment]
@hkaiser Reading more documentaiton the easiest way to produce the CITATION.cff file is to use the Python script provided by GitHub

`doi2cff init 10.5281/zenodo.5101081`

and for each release we would need to run 

`doi2cff update 10.5281/zenodo.5101081`

@msimberg Should we add that to the release instruction?

**Comment by hkaiser** [comment]
> @hkaiser Reading more documentation the easiest way to produce the CITATION.cff file is to use the Python script provided by GitHub
> 
> `doi2cff init 10.5281/zenodo.5101081`
> 
> and for each release we would need to run
> 
> `doi2cff update 10.5281/zenodo.5101081`
> 
> @msimberg Should we add that to the release instruction?

I think this should use `10.5281/zenodo.598202` instead as the above is a release-specific link that would have to be updated for each release.

**Comment by diehlpk** [comment]
> > @hkaiser Reading more documentation the easiest way to produce the CITATION.cff file is to use the Python script provided by GitHub
> > `doi2cff init 10.5281/zenodo.5101081`
> > and for each release we would need to run
> > `doi2cff update 10.5281/zenodo.5101081`
> > @msimberg Should we add that to the release instruction?
> 
> I think this should use `10.5281/zenodo.598202` instead as the above is a release-specific link that would have to be updated for each release.

Yes, we would have to update the Zenodo DOI with each release, since people would copy the DOI of the specific release. So people could indicate which version of HPX they used. For HPX itself, they would use the JOSS paper.

**Comment by hkaiser** [comment]
It depends whether we want to refer to the HPX software as such or whether we want to refer to the latest released version. I think that we should refer to the software as such, not to a particular version. However, I'm open to other people's suggestions.

**Comment by msimberg** [comment]
> @msimberg Should we add that to the release instruction?

Yes please, given that we do actually want to use the release-specific DOI. Is it possible to have multiple `cff` files, or is it standardized enough that anything not called `CITATION.cff` is most likely never going to be noticed by others/tooling?

**Comment by diehlpk** [comment]
@msimberg I think GitHub will only parse the CITATION.cff file. I am not sure about other tools.

**Comment by msimberg** [comment]
Let's discuss this in this week's meeting.

**Comment by hkaiser** [comment]
Here is what we've discussed today:
- we should go with the generic zenodo link in the CITATIONS file as the version specific link is generated only after the release was published on github, thus we wouldn't be able to have the version-specific url in the release itself
- we should remove any version from the CITATIONS file to keep it generic

This is probably fine as the text on the homepage says: 'Cite this repository` and not 'Cite this release' (or similar).

**Comment by diehlpk** [comment]
@hkaiser and @msimberg 

`doi2cff init 10.5281/zenodo.598202`

It seems that using the generic DOI, the Python scripts resolves that to the latest version.

**Comment by diehlpk** [comment]
I think, we need to generate the CIATION.cff file by hand for the generic version.

**Comment by msimberg** [comment]
> @hkaiser and @msimberg
> 
> `doi2cff init 10.5281/zenodo.598202`
> 
> It seems that using the generic DOI, the Python scripts resolves that to the latest version.

It seems like that might change. From https://help.zenodo.org/#versioning:
```
Where does the Concept DOI resolve to?

Currently the Concept DOI resolves to the landing page of the latest version of your record. This is not fully correct, and in the future we will change this to create a landing page specifically representing the concept behind the record and all of its versions.
```

Can we not just change the DOI in the generated file to use the generic one?

**Comment by msimberg** [comment]
@diehlpk we discussed this again at the last meeting and if possible, we'd prefer just using the generic DOI (we have a chicken-and-egg problem with the version specific DOI). If you agree, could you update the PR to reflect that?

**Comment by diehlpk** [comment]
@msimberg I will work on that next week.

**Comment by diehlpk** [comment]
@msimberg I opened #5553, and we should have the action to validate the cff file, since we might update the file in the near future.

**Comment by msimberg** [comment]
@diehlpk could you add the validator action here, please?

**Comment by diehlpk** [comment]
@msimberg I added the action, but it seems that I did something wrong. Would you or @aurianer mind to have a look? It seems you did that before.

File: .github/workflows/cff-validator.yml
Comment (hkaiser) [inline .github/workflows/cff-validator.yml:1]: This is missing License, copyright, and SPDX notice

File: CITATION.cff
Comment (hkaiser) [inline CITATION.cff:1]: This is missing license, copyright, and SPDX notice

File: .github/workflows/cff-validator.yml
Comment (msimberg) [inline .github/workflows/cff-validator.yml:2]:
```suggestion
  pull_request:
```

**Comment by msimberg** [comment]
@diehlpk please see https://github.com/STEllAR-GROUP/hpx/pull/5471/checks?check_run_id=3607697310#step:4:602.

Could also please squash the commits as the last thing before we merge this?

**Comment by diehlpk** [comment]
@msimberg Finally, the CITATION.cff is validated correctly. 

I would like this badge to the README file

[CITATION-cff](https://github.com/STEllAR-GROUP/hpx/actions/workflows/cff-validator.yml)

I am not sure how to do that? Could you please add that?

**Comment by hkaiser** [comment]
@diehlpk you can add the badge as:
```
|CFF|
```
and at the end of the README.rst file:
```
.. |CFF| image:: https://github.com/STEllAR-GROUP/hpx/actions/workflows/cff-validator.yml/badge.svg
    :target: https://github.com/STEllAR-GROUP/hpx/actions/workflows/cff-validator.yml
    :alt: Citations file for HPX
```

**Review by hkaiser — DISMISSED** [review]
LGTM, thanks!

**Comment by msimberg** [comment]
@diehlpk could you please squash the commits? Otherwise this looks good.

**Comment by diehlpk** [comment]
@msimberg Done

File: CITATION.cff
Comment (hkaiser) [inline CITATION.cff:23]: This has still an invalid character in it. Could you please fix that?
```suggestion
- given-names: Agustin
```

**Review by hkaiser — APPROVED** [review]
LGTM otherwise, thanks!

File: CITATION.cff
Comment (msimberg) [inline CITATION.cff:29]:
```suggestion
  family-names: Reverdell
```

Sorry for noticing this only now. Feel free to just apply this directly on github. Let's *actually* merge this after that.

Comment (aurianer) [reply CITATION.cff:29]: Thanks :)

Comment (diehlpk) [reply CITATION.cff]: Sorry for that, the Python script does a horrible jobs with the names. We need to be careful with updating the CFF file with each release.

**Comment by diehlpk** [comment]
@msimberg Have you added this procedure to the HPX release process?

We need to update the CFF file which each release or? At least we need to update the version number there or?

**Comment by msimberg** [comment]
> @msimberg Have you added this procedure to the HPX release process?
> 
> We need to update the CFF file which each release or? At least we need to update the version number there or?

You tell me ;)

I would assume so. Do I understand it correctly that we have a specific version in the file, but the doi is the generic one? Would you mind adding whatever steps are required to the release procedure?

**Pull request merged**
